### PR TITLE
Add support to x-codeSamples

### PR DESCRIPTION
**Pull request by lucasassisrosa**
## Description

This PR aims to fix the `x-codeSamples` [integration](https://redocly.com/docs/api-reference-docs/specification-extensions/x-code-samples). When `x-codeSamples` are provided, add as a variant for the target supported language in the API Explorer. At first it seems that the code was aiming to only display the custom code sample when `x-codeSamples` is provided. I assumed it's better to add the custom code samples to the existing generated code samples in the list of variants

## Motivation and Context

`x-codeSamples` redoc extension was partially supported and breaking the rendering if loaded in API Explorer. Continues #589 (@joecrop I tried to continue the implementation but the codebase has changed considerably)

## How Has This Been Tested?

Go to http://localhost:3000/petstore/add-pet and check API Explorer code samples. It reflects the OpenAPI spec now on `x-codeSamples`, following syntax highlighting

## Types of changes

- Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes if appropriate.
- [x] All new and existing tests passed.

### CODE REVIEW

File: packages/docusaurus-theme-openapi-docs/src/theme/ApiExplorer/CodeSnippets/languages.json
Comment (lucasassisrosa) [inline packages/docusaurus-theme-openapi-docs/src/theme/ApiExplorer/CodeSnippets/languages.json:1]: I didn't see any references to this file. Codebase was using `languageSet` from `@theme/ApiExplorer/CodeSnippets`

File: packages/docusaurus-theme-openapi-docs/src/theme/ApiExplorer/index.tsx [hunk 1]
```diff
@@ -30,7 +30,7 @@ function ApiExplorer({
       {item.method !== "event" && (
         <CodeSnippets
           postman={postman}
-          codeSamples={(item as any)["x-code-samples"] ?? []}
+          codeSamples={(item as any)["x-codeSamples"] ?? []}
```
Comment (lucasassisrosa) [inline packages/docusaurus-theme-openapi-docs/src/theme/ApiExplorer/index.tsx:33]: `x-code-samples` is the deprecated one

File: packages/docusaurus-theme-openapi-docs/src/theme/ApiExplorer/CodeSnippets/index.tsx [hunk 2]
```diff
@@ -335,11 +354,13 @@ function CodeSnippets({ postman, codeSamples }: Props) {
...
               >
-                {lang.variants.map((variant) => {
+                {lang.variants.map((variant, index) => {
                   return (
                     <CodeTab
                       value={variant.toLowerCase()}
-                      label={variant.toUpperCase()}
+                      label={
+                        lang.labels ? lang.labels[index] : variant.toUpperCase()
+                      }
```
Comment (lucasassisrosa) [inline packages/docusaurus-theme-openapi-docs/src/theme/ApiExplorer/CodeSnippets/index.tsx:367]: labels might be useful for future support of different variant values and labels

File: demo/examples/petstore.yaml [hunk 3]
```diff
@@ -147,6 +147,7 @@ paths:
               Console.WriteLine(response.getRawResponse());
             }
         - lang: PHP
+          label: Custom
```
Comment (lucasassisrosa) [inline demo/examples/petstore.yaml:150]: just to leave an example of custom label

File: packages/docusaurus-theme-openapi-docs/src/theme/ApiExplorer/CodeSnippets/languages.json
Comment (sserrata) [reply packages/docusaurus-theme-openapi-docs/src/theme/ApiExplorer/CodeSnippets/languages.json:1]: You're correct, I included that file as a maintainer reference for what languages/variants are available.

File: packages/docusaurus-theme-openapi-docs/src/theme/ApiExplorer/CodeSnippets/index.tsx (hunk 2 above)
Comment (sserrata) [reply packages/docusaurus-theme-openapi-docs/src/theme/ApiExplorer/CodeSnippets/index.tsx:367]: Makes sense...will these labels effectively override the defaults if passed via docusaurus config to a "generated" language?

Comment (lucasassisrosa) [reply packages/docusaurus-theme-openapi-docs/src/theme/ApiExplorer/CodeSnippets/index.tsx:367]: yes they will, but they're not right now. Right now they're just overriding the defaults for code samples

Comment (sserrata) [reply packages/docusaurus-theme-openapi-docs/src/theme/ApiExplorer/CodeSnippets/index.tsx:367]: Agreed...also, there is another issue with generated code snippet - if you try to rearrange the order it leads to some errors. Additionally, I had hoped to allow users to specify which variants they wanted to render, but that needs additional work.

Comment (lucasassisrosa) [reply packages/docusaurus-theme-openapi-docs/src/theme/ApiExplorer/CodeSnippets/index.tsx:367]: @sserrata with " rearrange the order" you mean in CodeSnippets `languageSet` or in Docusaurus Config `languageTabs`?

Comment (lucasassisrosa) [reply packages/docusaurus-theme-openapi-docs/src/theme/ApiExplorer/CodeSnippets/index.tsx]: @sserrata I have updated the code samples implementation to keep it split from variants, having two CodeTabs. Please let me know if that's accurate. I also though about hiding the Variants (auto generated) tabs if there are CodeSamples but I don't know if that's the path the framework wants to move forward

https://github.com/PaloAltoNetworks/docusaurus-openapi-docs/pull/697/commits/27f63767b73124d5a276be77019ea7c6348b5232

Comment (sserrata) [reply packages/docusaurus-theme-openapi-docs/src/theme/ApiExplorer/CodeSnippets/index.tsx]: > @sserrata with " rearrange the order" you mean in CodeSnippets languageSet or in Docusaurus Config languageTabs?

Docusaurus config languageTabs, although that issue has been around for some time and is not related to your changes.

**Comment by sserrata** [comment]
Hi @lucasassisrosa, I think this PR accomplishes the goal of supporting `x-codeSamples`. I think it's ready to merge as-is but I plan to follow up to address the issues pertaining to:

* allowing users to specify which variants to include
* allowing users to customize the order languages are rendered
* allowing users to completely disable/hide the snippets component
* allowing users to only show custom languages (i.e. hide generated)

**Review by sserrata — APPROVED** [review]

**Pull request merged**